### PR TITLE
Add NSData.toByteString()

### PR DESCRIPTION
**Pull request by Egorand**

### CODE REVIEW

File: okio/build.gradle [hunk 1]
```diff
@@ -66,13 +66,25 @@ kotlin {
     nativeTest {
       dependsOn commonTest
     }
+    appleMain {
```
Comment (Egorand) [inline okio/build.gradle]: Not sure if "apple" is the right term for iOS + macOS

File: okio/src/appleMain/kotlin/okio/ByteString.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+package okio
+
+import kotlinx.cinterop.addressOf
+import kotlinx.cinterop.usePinned
+import platform.Foundation.NSData
+import platform.posix.memcpy
+
+fun NSData.toByteString(): ByteString {
+  val data = this
+  return ByteString(ByteArray(data.length.toInt()).apply {
```
Comment (JakeWharton) [inline okio/src/appleMain/kotlin/okio/ByteString.kt]: Why is `toInt()` needed? Is it unsigned normally?

Comment (Egorand) [reply okio/src/appleMain/kotlin/okio/ByteString.kt]: Yeah,
>  inferred type is NSUInteger /* = ULong */

File: okio/src/appleMain/kotlin/okio/ByteString.kt [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
```
Comment (swankjesse) [inline okio/src/appleMain/kotlin/okio/ByteString.kt]: NEAT

File: okio/src/appleTest/kotlin/okio/AppleByteStringTest.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okio
+
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import platform.Foundation.NSData
```
Comment (swankjesse) [inline okio/src/appleTest/kotlin/okio/AppleByteStringTest.kt]: COOOL

**Comment by JakeWharton** [comment]
Darwin?

**Comment by swankjesse** [comment]
Darwin’s mostly a coincidence here. The AppKit / UiKit / NextStep APIs could target Linux or Windows, and I suspect there’s code in iTunes and Safari for Windows that does so.

**Comment by swankjesse** [comment]
oooooh http://cocotron.org/

**Comment by JakeWharton** [comment]
You could use that argument against Linux because it runs on Windows or Windows because it runs on Linux.

**Comment by Egorand** [comment]
Are we likely to use any types outside of `Foundation`? Would `foundation` make sense then?

**Comment by swankjesse** [comment]
I think Apple is the best name.

**Pull request merged**